gateway_to_backend/protocol_buffers_files/otap_message.proto

```diff
@@ -24,11 +24,23 @@ message GetScratchpadStatusResp {
 }
 
 message UploadScratchpadReq {
+    message ChunkInfo {
```

Comment:
Maybe more of a question than any actual comment? Is there some reason for nested typing here? There seem to be couple of enums defined as nested, but otherwise we mostly define message types as "top level" entities. Does it cause some compability issues, if we later decide to use this type also in some other message (like in response what was already speculated) and would then need to extract this from here to top level?

Reply:
I don't know if you can "extract" the message, but at least you can reuse it outside. Just for info : 
"If you want to reuse this message type outside its parent message type, you refer to it as _Parent_._Type_:
message SomeOtherMessage {
  SearchResponse.Result result = 1;
}"
I'd also prefer to avoid nesting if possible, seems to me more readable to have it outside (as a protobuf-beginner)

Reply:
I think that there was some issue with enum inside the message, if you want to use it somewhere else, but I cannot remember it anymore. Might have been some client library implementation issue too.

Reply:
Yeap, my point was more related to the general pattern that we use elsewhere in this API's protobufs, seems that most of the types are anyways defined at the top level (probably mainly due to the fact that they are used in different messages and there is no clear "parent" for the type.

Reply:
Yes exactly. I think it is the first one where nested object make sense, as it shouldn't be used anywhere else. But if you prefer, I can put it outside.
For me the nesting seems a bit cleaner, but no strong opinion :-)

Reply:
Yeap, no strong opinion either. We can go with nested type here, as the child - parent relation is clear in this case as you said.